### PR TITLE
Bump min compiler version to 515.1627

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

Here we go again.

1624 fixed some bugs with `::`, which we're starting to make more use of. 

image

Case in point: This test run https://github.com/tgstation/tgstation/actions/runs/8147514803/job/22268415319?pr=81726

But then we need the fixes for 1624 so we go to 1626. 

But also 1627 has compiler improvements so might as well

### CODE REVIEW

File: code/__byond_version_compat.dm [hunk 1]
```diff
@@ -2,11 +2,11 @@
 
 //Update this whenever you need to take advantage of more recent byond features
 #define MIN_COMPILER_VERSION 515
-#define MIN_COMPILER_BUILD 1621
+#define MIN_COMPILER_BUILD 1624
 #if (DM_VERSION < MIN_COMPILER_VERSION || DM_BUILD < MIN_COMPILER_BUILD) && !defined(SPACEMAN_DMM)
 //Don't forget to update this part
 #error Your version of BYOND is too out-of-date to compile this project. Go to https://secure.byond.com/download and update.
-#error You need version 515.1621 or higher
+#error You need version 515.1624 or higher
```
Comment (san7890) [inline code/__byond_version_compat.dm:9]:
```suggestion
#error You need version [MIN_COMPILER_VERSION].[MIN_COMPILER_BUILD] or higher
```

I wonder if this works so we don't have ot keep bumping up this string (no issue if it doesn't)

**Comment by san7890** [comment]
image

that can't be good

**Comment by san7890** [comment]
image

also why was the test meant to ensure that that the test run would fail loudly skip? issue for a different pr but i'll bring it up now because it just happened

**Comment by MrMelbert** [comment]
1624 is a bad version too? lol

**Comment by san7890** [comment]
Why are we even testing for 1624 if the standard version is 1630? Should we just do away with this test?

**Comment by MrMelbert** [comment]
> Why are we even alternate-testing for 1624 if the bare-minimum `dependences.sh` version is 1630 (with the latest being 1632)? Should we just do away with this test if it shrimply does not work?
> 
> The alternate test was only there to test for 515 beta versions when main-line CI was still stuck on 514. Now that we've fully migrated to 515.1630 for everything, it's pretty pointless to keep running tests on incompatible backwards BYOND versions. Will save CI runners too.

We put alternate test on min compiler version because we already had one case in which the min compiler version was broken without anyone noticed #81183 

I guess I'll bump it to 1630

File: code/__byond_version_compat.dm (hunk 1 above)
Comment (MrStonedOne) [reply code/__byond_version_compat.dm]: it doesn't or we would already

**Comment by MrMelbert** [comment]
> Why are we even alternate-testing for 1624 if the bare-minimum `dependences.sh` version is 1630 (with the latest being 1632)? Should we just do away with this test if it shrimply does not work?

`dependences.sh` is what CI runs on, not the minimum version

**Comment by MrStonedOne** [comment]
we should only have min compiler version be as high as it needs to be **and no higher**.

we should sometimes bump the dependencies.sh version to a higher number than the min support compiler if it has speed improvements to help with the time to run ci.

Alt tests lets us have it both ways.

**Comment by san7890** [comment]
Alright, that makes sense. Keep in mind that having the minimum be desynced from dependencies.sh (which is what TGS and everything reads as it is the "final authority"™️) only works as long as the minimum works (no duh moment) but nothing grinds my gears more than spontaneous CI combustion so I will always be more on the liberal side of dialing up the version if it veritably does not work

**Comment by LemonInTheDark** [comment]
> we should only have min compiler version be as high as it needs to be **and no higher**.
> 
> we should sometimes bump the dependencies.sh version to a higher number than the min support compiler if it has speed improvements to help with the time to run ci.

Agree with first part
dependencies.sh should be semi regularly bumped to the latest stable byond version IMO. It prevents large (unknown) errors caused by the engine from stacking up and needing to all be cleared away in one step

**Comment by LemonInTheDark** [comment]
That or have an alt test on latest idk

**Pull request merged**